sources/VT100Grid.m

```diff
@@ -405,7 +405,7 @@ - (int)resetWithLineBuffer:(LineBuffer *)lineBuffer
     }
     self.cursor = VT100GridCoordMake(0, 0);
 
-    [self setCharsFrom:VT100GridCoordMake(0, preserveCursorLine ? 1 : 0)
+    [self setCharsFrom:VT100GridCoordMake(0, preservePromptLines)
```

Comment:
Agreed, fix this in line 395 above - add check for preservePromptLines < size_.height.